include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2179,7 +2179,7 @@ class binary_reader
                     for (auto i : dim)
                     {
                         result *= i;
-                        if (result == 0) // because dim elements shall not have zeros, result = 0 means overflow happened
+                        if (result == 0 || result == string_t::npos) // because dim elements shall not have zeros, result = 0 means overflow happened; it also can't be string_t::npos
```

Comment:
Can you please also add a reason why it can't be npos?

Reply:
done